codegen/src/main/java/software/amazon/awssdk/codegen/poet/auth/scheme/DefaultAuthSchemeParamsSpec.java

```diff
@@ -77,16 +79,28 @@ private MethodSpec constructor() {
 
         if (authSchemeSpecUtils.generateEndpointBasedParams()) {
             parameters().forEach((name, model) -> {
-                if (authSchemeSpecUtils.includeParam(model, name)) {
+                if (authSchemeSpecUtils.includeParam(name)) {
                     String fieldName = authSchemeSpecUtils.paramMethodName(name);
-                    b.addStatement("this." + fieldName + " = builder." + fieldName);
+                    boolean isRequired = isParamRequired(model);
+                    if (isRequired) {
+                        b.addStatement(String.join(fieldName, "this.", " = $T.paramNotNull(builder.", ", \"", "\")"),
+                                       Validate.class);
+                    } else {
+                        b.addStatement("this." + fieldName + " = builder." + fieldName);
+                    }
+
                 }
             });
         }
 
         return b.build();
     }
 
+    private boolean isParamRequired(ParameterModel model) {
+        Boolean isRequired = model.isRequired();
+        return (isRequired != null && isRequired) || model.getDefault() != null;
```

Comment:
nit: There is "Parameters with defaults MUST also be marked as required.", so `model.getDefault()` check isn't technically needed.

Reply:
I didn't know that, where is this 

> There is "Parameters with defaults MUST also be marked as required.",

coming from?

I will remove the extra check.

Reply:
Added back, some of the tests file have default but not required.